### PR TITLE
#10052 - Use try-with-resources to close "FileInputStream gzippedIn =…

### LINKED ISSUE #10052: Use try-with-resources to close "FileInputStream gzippedIn = new FileInputStream(f);" in FileUtil.java in method determineFileType(...)

### PR DESCRIPTION
**Pull request by Monoradioactivo**
**What this PR does / why we need it**:

**Which issue(s) this PR closes**:

Closes #10052 

**Special notes for your reviewer**:

**Suggestions on how to test this**:

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java [hunk 1]
```diff
@@ -501,24 +501,30 @@ public static String determineFileType(File f, String fileName) throws IOExcepti
...
-            } catch (IOException ioex) {
-                if (uncompressedIn != null) {
-                    try {uncompressedIn.close();} catch (IOException e) {}
+            try (FileInputStream gzippedIn = new FileInputStream(f)) {
+                // (new FileInputStream() can throw a "file not found" exception;
+                // however, if we've made it this far, it really means that the
+                // file does exist and can be opened)
+                InputStream uncompressedIn = null;
+                try {
+                    uncompressedIn = new GZIPInputStream(gzippedIn);
```
Comment (qqmyers) [inline src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java:510]: It wasn't in the issue but would you be willing to do the same thing to this stream? I think you should be able to put this in line 504 in the same try.

Comment (Monoradioactivo) [reply src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java]: I've updated the PR

**Review by qqmyers — APPROVED** [review]
Awesome - thanks!

**Comment by pdurbin** [comment]
Hmm. The fork was deleted. Oh well.

**Pull request closed without merging**